model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java

```diff
@@ -992,7 +992,7 @@ private List<Predicate> predicates(Map<String, String> attributes, Root<UserEnti
                     Join<UserEntity, UserAttributeEntity> attributesJoin = root.join("attributes", JoinType.LEFT);
 
                     attributePredicates.add(builder.and(
-                            builder.equal(builder.lower(attributesJoin.get("name")), key.toLowerCase()),
+                            builder.equal(attributesJoin.get("name"), key),
                             builder.equal(builder.lower(attributesJoin.get("value")), value.toLowerCase())));
```

Comment:
Perhaps you did lower in the other method to get consistent with this one?

Reply:
Exactly.